morpheus/controllers/vector_db_controller.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+import typing
+from abc import ABC
+from abc import abstractmethod
+
+
+class VectorDBController(ABC):
+    """
+    Abstract class for vector document store implementation.
+    """
+
+    _mutex = threading.Lock()
+
+    @abstractmethod
+    def insert(self, name: str, data: typing.Any, **kwargs):
```

Comment:
I think it would be good to define the supported types at the interface level for `data`

Reply:
Added data types in interface API